pkg/destroy/aws/aws.go

```diff
@@ -1832,6 +1832,43 @@ func deleteS3(session *session.Session, arn arn.ARN, logger logrus.FieldLogger)
 	}
 	logger.Debug("Emptied")
 
+	var lastError error
+	err = client.ListObjectVersionsPagesWithContext(aws.BackgroundContext(), &s3.ListObjectVersionsInput{
+		Bucket:  aws.String(arn.Resource),
+		MaxKeys: aws.Int64(1000),
+	}, func(page *s3.ListObjectVersionsOutput, lastPage bool) bool {
+		var deleteObjects []*s3.ObjectIdentifier
+		for _, deleteMarker := range page.DeleteMarkers {
+			deleteObjects = append(deleteObjects, &s3.ObjectIdentifier{
+				Key:       aws.String(*deleteMarker.Key),
+				VersionId: aws.String(*deleteMarker.VersionId),
+			})
+		}
+		for _, version := range page.Versions {
+			deleteObjects = append(deleteObjects, &s3.ObjectIdentifier{
+				Key:       aws.String(*version.Key),
+				VersionId: aws.String(*version.VersionId),
+			})
+		}
+		_, err := client.DeleteObjects(&s3.DeleteObjectsInput{
```

Comment:
we need to skip deleting objects, when the delete objects is empty list. @jhixson74